code/modules/integrated_electronics/subtypes/manipulation.dm

```diff
@@ -203,4 +202,171 @@
+	extended_desc = "The circuit accepts a reference to creature,who needs to be shocked. It can shock target on adjastent tiles.."
+	w_class = ITEMSIZE_TINY
+	complexity = 10
+	inputs = list("target" = IC_PINTYPE_REF)
+	outputs = list()
+	activators = list("shock" = IC_PINTYPE_PULSE_IN)
+	spawn_flags = IC_SPAWN_RESEARCH
+	power_draw_per_use = 50
+
+/obj/item/integrated_circuit/manipulation/shocker/do_work()
+	..()
+	var/turf/T = get_turf(src)
+	var/atom/movable/AM = get_pin_data_as_type(IC_INPUT, 1, /mob/living)
+	if(!istype(AM,/mob/living)) //Invalid input
+		return
+	var/mob/living/M = AM
+	if(!M.Adjacent(T))
+		return //Can't reach
+	M << "<span class='danger'>You feel a sharp shock!</span>"
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	s.set_up(3, 1, M)
+	s.start()
+	M.Weaken(10)
+
+/obj/item/integrated_circuit/manipulation/grabber
+	name = "grabber"
+	desc = "circuit with it's own inventory for small/medium items ,used to grab and store things."
+	icon_state = "grabber"
+	extended_desc = "The circuit accepts a reference to thing to be grabbed. It can store up to 10 things.Modes:1 for grab. 0 for eject the first thing. -1 for eject all"
+	w_class = ITEMSIZE_NORMAL
+	size = 3
+
+	complexity = 10
+	inputs = list("target" = IC_PINTYPE_REF,"mode" = IC_PINTYPE_NUMBER)
+	outputs = list("first" = IC_PINTYPE_REF, "last" = IC_PINTYPE_REF, "amount" = IC_PINTYPE_NUMBER)
+	activators = list("pulse in" = IC_PINTYPE_PULSE_IN,"pulse out" = IC_PINTYPE_PULSE_OUT)
+	spawn_flags = IC_SPAWN_RESEARCH
+	power_draw_per_use = 50
+/*
+/obj/item/integrated_circuit/manipulation/thrower/New()
+	processing_objects |= src
+
+/obj/item/integrated_circuit/manipulation/thrower/Destroy()
+	processing_objects -= src
+
+/obj/item/integrated_circuit/manipulation/thrower/process()
+	set_pin_data(IC_OUTPUT, 1, weakref(contents[1]))
+	set_pin_data(IC_OUTPUT, 2, weakref(contents[contents.len]))
+	set_pin_data(IC_OUTPUT, 3, src.contents.len)
+	push_data()
+*/
+/obj/item/integrated_circuit/manipulation/grabber/do_work()
+	var/turf/T = get_turf(src)
+	var/atom/movable/AM = get_pin_data_as_type(IC_INPUT, 1, /obj/item)
+	var/mode = get_pin_data(IC_INPUT, 2)
+	if(mode == 1)
+		if(AM.Adjacent(T))
+			if(contents.len < 10)
+				if(istype(AM,/obj/item))
+					var obj/item/A = AM
+					if(A.w_class < ITEMSIZE_NORMAL)
+						AM.loc = src
+				else if(istype(AM,/mob))
+					var mob/A = AM
+					if(istiny(A))
+						AM.loc = src
+	if(mode == 0)
+		if(contents.len)
+			var/obj/item/U = contents[1]
+			U.loc = T
+	if(mode == -1)
+		if(contents.len)
+			var/obj/item/U
+			for(U in contents)
+				U.loc = T
+	set_pin_data(IC_OUTPUT, 1, weakref(contents[1]))
+	set_pin_data(IC_OUTPUT, 2, weakref(contents[contents.len]))
+	set_pin_data(IC_OUTPUT, 3, src.contents.len)
+	push_data()
+	activate_pin(2)
+
+/obj/item/integrated_circuit/manipulation/thrower
+	name = "thrower"
+	desc = "compact launcher to throw things from inside or nearby tiles"
```

Comment:
Add a capital A at the start or capitalize the C.  Also need a period at the end.